.github/workflows/todoapp_pipeline.yaml

```diff
@@ -0,0 +1,25 @@
+name: Continuos Integration
+on:
+  push:
+    paths-ignore:
+      - 'README.md'
+      - '.gitignore'
+      - 'documentation'
+  pull_request:
+    paths-ignore:
+      - 'README.md'
+      - '.gitignore'
+      - 'documentation'
+  schedule:
+    - cron: '40 8 * * *' 
+jobs:
+  build:
+    name: Build and test
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout files 
+        uses: actions/checkout@v2
+      - name: Build docker
+        run: docker build --target testing --tag todo-app:testing .
+      - name: Run tests
+        run: docker run --env-file ./.env.test todo-app:testing
```

Comment:
Since you have a compose file that includes running tests, you could use that instead of docker build and run commands.

Something like this should work:
`docker compose up testing --exit-code-from testing`

You need the `--exit-code-from` part to say that if the `testing` service fails, then the `docker compose` command should exit with that exit code too.